lib/FreeRTOS-Plus-TCP/source/FreeRTOS_DNS.c

```diff
@@ -446,7 +449,13 @@ TickType_t xIdentifier = 0;
 	/* Generate a unique identifier. */
 	if( 0 == ulIPAddress )
 	{
-		xIdentifier = ( TickType_t )ipconfigRAND32( );
+		/* DNS identifiers are 16-bit. */
+		xIdentifier = ( TickType_t )( ipconfigRAND32( ) & 0xffff );
+		/* An identifier with value 0 will be treated as: "no identifier". */
```

Comment:
We unfortunately have an ambiguity wherein zero is both a valid random number and a possible error return. Fixing that is a pain, since ipconfigRAND32 tends to be #defined to a port specific implementation.

In any case, we don't want to issue a request with a non-random random transaction ID.

Reply:
Elsewhere in the library, the result of `ipconfigRAND32()` is tested against 0 with the following comment ( not mine ) :

`	xDHCPData.ulTransactionId = ipconfigRAND32();`

`	/* Check for random number generator API failure. */`
`	if( 0 != xDHCPData.ulTransactionId )`
`	{`

Today I **did** have a complaint that DHCP didn't work. The user's implementation of `ipconfigRAND32` returned zero at the first call.

( I always emphasize to users that the first value of the seed must be random )

We can not require that `ipconfigRAND32()` never returns 0, or let things go wrong when ocasionally 0 is returned.

It is possible to call `ipconfigRAND32()` a few times until it returns a non-zero value?